framework/aster-frame/src/sync/condvar.rs

```diff
@@ -0,0 +1,109 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use alloc::sync::Arc;
+use core::{
+    sync::atomic::{AtomicBool, Ordering},
+    time::Duration,
+};
+
+use super::{Mutex, MutexGuard, WaitQueue};
+
+/// A condition variable.
+pub struct Condvar {
+    wait_queue: WaitQueue,
+}
+
+impl Condvar {
+    /// Creates a new condition variable.
+    pub const fn new() -> Self {
+        Condvar {
+            wait_queue: WaitQueue::new(),
+        }
+    }
+
+    /// Waits for the condition variable to be signaled or broadcasted.
+    pub fn wait<'a, T>(&self, guard: MutexGuard<'a, T>) -> MutexGuard<'a, T> {
+        guard.mutex.unlock();
+        let should_wait = || {
+            // Check the condition for which we're waiting.
+            // Set to dummy value for now.
+            None::<T>
+        };
```

Comment:
The condition is None::<T> so the waiting thread will never be woken up. I believe there should be another value to indicate whether the thread is notified.

Reply:
Thank you for pointing that. I solve this by add a `VecDeque` of `AtomicBool` to store and detect the notify state of threads.